app/level-editor/panels/viewport_panel.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2020, Dmitry Shilnenkov
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "viewport_panel.h"
+#include "editor_state.h"
+
+#include "ge/debug/profile.h"
+#include "ge/gui/gui.h"
+#include "ge/renderer/framebuffer.h"
+
+#include <imgui.h>
+
+namespace LE {
+
+ViewportPanel::ViewportPanel(GE::Shared<EditorState> editor_state)
+    : m_editor_state{std::move(editor_state)}
+{
+    GE_PROFILE_FUNC();
+}
+
+void ViewportPanel::onGuiRender()
+{
+    GE_PROFILE_FUNC();
+
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2{0, 0});
+
+    if (ImGui::Begin("Viewport")) {
+        bool is_vp_focused = ImGui::IsWindowFocused();
+        bool is_vp_inactive = !is_vp_focused || !ImGui::IsWindowHovered();
+        GE::Gui::blockEvents(is_vp_inactive);
+
+        ImVec2 vp_panel_size = ImGui::GetContentRegionAvail();
+        auto framebuffer_tex = m_editor_state->framebuffer()->getColorAttachmentID();
+        auto* fb_texture = reinterpret_cast<ImTextureID>(framebuffer_tex);
```

Comment:
`ImTextureID` -> `void*` looks weird
```suggestion
        ImTextureID fb_texture = reinterpret_cast<ImTextureID>(framebuffer_tex);
```